docs/changelog/4_x.rst

```diff
@@ -0,0 +1,454 @@
+        }
+
+        type Item {
+            required sku: str;
+            required description: str;
+        }
+
+        type Cart {
+            required customer: Customer;
+            multi items: Item {
+                quantity: int32;
+            };
+
+            access policy customer_has_full_access
+                allow all
+                using (global current_customer.id ?= .customer.id);
+        }
+    }
+
+
+Here's an example query using the TypeScript client:
+
+.. code-block:: typescript
+
+    import { createClient } from "edgedb";
+
+    declare const tokenFromAuthServer: string;
+    const client = createClient()
+      .withGlobals({
+        "ext::auth::client_token": tokenFromAuthServer
+      });
+
+    const carts = await client.query(`select Cart { * };`);
+
+
+
+pgcrypto
+--------
+
+We've added ``pgcrypto`` to our extensions. This exposes ``digest``, ``hmac``,
+``gen_salt`` and ``crypt`` functions for your hashing, ecrypting and salting
+needs.
+
+.. code-block:: edgeql-repl
+
+  edgedb> select ext::pgcrypto::digest('encrypt this', 'sha1');
+  {b'\x05\x82\xd8YLF\xe7\xd4\x12\x91\n\xdb$\xf1!v\xf9\xd4\x89\xc4'}
+  edgedb> select ext::pgcrypto::gen_salt('md5');
+  {'$1$FjNlXgX7'}
+
+
+pg_trgm
+-------
+
+This extension provides a bunch of functionality used to determine string
+similarity, which makes it a good text search alternative for some use cases:
+
+edgedb> with x := {'hello world', 'word hero', 'help the world'}
+....... select res := (x, ext::pg_trgm::word_similarity(x, 'hello world'))
+....... order by res.1 desc;
+{('hello world', 1), ('help the world', 0.5), ('word hero', 0.35714287)}
+
+
+
+Additional changes
+==================
+
+EdgeQL
+------
+
+* Add ``anyobject`` pseudo type.
+  (:eql:gh:`#5961`)
+
+* Add new style of ``if``/``then``/``else`` syntax.
+  (:eql:gh:`#6074`)
+
+  Many people find it more natural to write "if ... then .. else ..." for
+  conditional expressions because it mirrors the conditional statement from
+  other familiar programming languages.
+
+  .. code-block:: edgeql-repl
+
+    edgedb> select if count(Object) > 0 then 'got data' else 'no data';
+    {'got data'}
+
+* Support conditional DML.
+  (:eql:gh:`#4437`)
+
+  It can be useful to be able to create, update or delete different objects
+  based on some condition:
+
+  .. code-block:: edgeql
+
+    with
+      name := <str>$0,
+      admin := <bool>$1
+    select if admin then (
+        insert AdminUser { name := name }
+    ) else (
+        insert User { name := name }
+    )
+
+  A different use-case of conditional DML is using a :eql:op:`coalesce`
+  operator to express things like "select or insert if missing":
+
+  .. code-block:: edgeql
+
+    select (select User filter .name = 'Alice') ??
+           (insert User { name := 'Alice' });
+
+* Add ``contains`` for JSON so that it can be used with ``pg::gin`` index.
+  (:eql:gh:`#5703`)
+
+* Add :eql:func:`to_bytes` and its counterpart :eql:func:`to_str`.
```

Comment:
We use `to_str` for a lot of formatting. Do we have that in `to_bytes`? I mean perhaps `to_bytes` is not a counterpart, it's a way to encode strings and serialize other types in some way, no?

Reply:
Agree, it's more of a "sibling" than it is a "counterpart"